back/index.js

```diff
@@ -1,14 +1,36 @@
 const express = require('express');
 const mongoose = require('mongoose');
+const bodyParser = require('body-parser');
 
 const Product = require('./models/product');
 
-const App = express();
-const connect = mongoose.connect('mongodb://localhost:27017/express-react-shop');
+const app = express();
+const port = 5001;
 
-App.get('/products', async (req, res) => {
-  const products = await Product.find();
-  res.json(products);
-});
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(bodyParser.json());
 
-App.listen(3000, () => console.log('The server is started'));
+mongoose.connect(
+  'mongodb://localhost:27017/express-react-shop',
+  { useNewUrlParser: true }
+);
+
+app.route('/products')
+  .get(async (req, res) => {
+    const products = await Product.find();
+    res.json(products);
+  })
+  .post(async (req, res) => {
+    const product = await Product.create(req.body.product);
+    res.json(product);
+  });
+
+app.route('/products/:id')
+  .get(async (req, res) => {
+    const product = await Product.findById(req.params.id);
```

Comment:
For this route, let's have a `404` message if the product is not found. This will be very important for the frontend too.